Source/List/ListCellSizeController.swift

```diff
@@ -40,8 +41,29 @@ internal final class ListCellSizeController {
     case let .explicit(size):
       return size
     case .relative:
-      assertionFailure("Relative sizing is not supported for supplementary views")
-      return autolayoutSize(for: cellModel, constrainedTo: sizeConstraints)
+      guard
+        let indexPath = indexPath
+      else {
+        assertionFailure(
+          "An indexPath should always be provided for .relative"
+        )
+        return .zero
+      }
+      guard
+        let size = delegate?
+        .sizeController(
```

Comment:
This is using the same delegate method for the cell model that the supplementary view is supporting which can lead to bugs if the system expects the actual cell to be used here.  Another (optional?) delegate method is needed that would be `sizeController(_:,sizeForSupplementaryView:,at:,constrainedTo:)`

Reply:
I've updated the PR to add an additional method for implementing supplementary view size.